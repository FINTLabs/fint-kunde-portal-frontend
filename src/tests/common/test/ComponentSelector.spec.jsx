import React from "react";
import { create } from "react-test-renderer";
import ComponentSelector from "../../../common/test/ComponentSelector";

describe("Component selector", () => {
  const component = create(
    <ComponentSelector
      name="test"
      value="test"
      components={[
        {
          dn: "cn=comp1",
          basePath: "/comp1",
          desciption: "Comp 1"
        },
        {
          dn: "cn=comp2",
          basePath: "/comp2",
          desciption: "Comp 2"
        }
      ]}
      handleChange={() => {}}
    />
  );

  it("should match snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
