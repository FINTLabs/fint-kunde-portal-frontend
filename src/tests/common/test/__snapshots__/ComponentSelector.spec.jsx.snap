// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component selector should match snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
      data-shrink={true}
      htmlFor="test"
    >
      Komponent
      <span
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
        â€‰
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="select-test"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          
        </div>
        <input
          id="test"
          name="test"
          type="hidden"
          value="test"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
